distribution/openrct2.d.ts

```diff
@@ -1582,16 +1583,23 @@ declare global {
         direction: Direction;
         object: number;
         primaryColour: number;
-        secondaryColour: number;
+        secondaryColour: number; // text colour if element is banner
         tertiaryColour: number;
-        bannerIndex: number | null;
         sequence: number;
+        readonly bannerIndex: number | null; // writing to bannerIndex is @deprecated and results in uncontrolled behaviour
+        text: string | null; // banner text if element is banner
```

Comment:
```suggestion
        /*
         * If the object is a banner, this will be the displayed text.
         */
        text: string | null;
        
        /*
         * @deprecated The banner index will result in uncontrolled behaviour.
         */
        readonly bannerIndex: number | null; 
```
Maybe good to move the deprecated properties to the bottom to keep them a bit out of sight.

Reply:
only the writable part is deprecated; reading is still supported and very much needed, e.g. for actions

Reply:
Ah then I'd suggest something like
```
@deprecated Setting the banner index may result in uncontrolled behaviour.
```